File: sdks/go/pkg/models/shared/devicepinners.go
package shared

type DevicePinnersOrderByEnum string

const (
	DevicePinnersOrderByEnumID DevicePinnersOrderByEnum = "id"
)

type DevicePinnersOrderDirectionEnum string

const (
	DevicePinnersOrderDirectionEnumAsc  DevicePinnersOrderDirectionEnum = "asc"
	DevicePinnersOrderDirectionEnumDesc DevicePinnersOrderDirectionEnum = "desc"
)

// DevicePinnersOrder
// The order in which a pagination is ordered
type DevicePinnersOrder struct {
	By        *DevicePinnersOrderByEnum        `json:"by,omitempty"`
	Direction *DevicePinnersOrderDirectionEnum `json:"direction,omitempty"`
}

// DevicePinners
// A list of device pins
type DevicePinners struct {
	Entries    []DevicePinner       `json:"entries,omitempty"`
	Limit      *int64               `json:"limit,omitempty"`
	NextMarker *int64               `json:"next_marker,omitempty"`
	Order      []DevicePinnersOrder `json:"order,omitempty"`
}

File: sdks/ts/src/sdk/models/shared/devicepinners.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { DevicePinner } from "./devicepinner";


export enum DevicePinnersOrderByEnum {
    Id = "id"
}

export enum DevicePinnersOrderDirectionEnum {
    Asc = "asc",
    Desc = "desc"
}


// DevicePinnersOrder
/** 
 * The order in which a pagination is ordered
**/
export class DevicePinnersOrder extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=by" })
  by?: DevicePinnersOrderByEnum;

  @SpeakeasyMetadata({ data: "json, name=direction" })
  direction?: DevicePinnersOrderDirectionEnum;
}


// DevicePinners
/** 
 * A list of device pins
**/
export class DevicePinners extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: DevicePinner })
  entries?: DevicePinner[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=next_marker" })
  nextMarker?: number;

  @SpeakeasyMetadata({ data: "json, name=order", elemType: DevicePinnersOrder })
  order?: DevicePinnersOrder[];
}
